packages/sil-governance/src/routes/create-proposal/components/ProposalForm.tsx

```diff
@@ -1,54 +1,70 @@
-import { ProposalOptions, ProposalType } from "@iov/bns-governance";
+import { ElectionRule } from "@iov/bns";
+import { Governor, ProposalOptions, ProposalType } from "@iov/bns-governance";
 import { Block, Button, Form, FormValues, Typography, useForm } from "medulas-react-components";
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import * as ReactRedux from "react-redux";
 
 import { sendSignAndPostRequest } from "../../../communication/signandpost";
 import { getBnsConnection } from "../../../logic/connection";
 import { RootState } from "../../../store/reducers";
-import DescriptionField from "./DescriptionField";
+import CommitteeSelect from "./CommitteeSelect";
+import DescriptionField, { DESCRIPTION_FIELD } from "./DescriptionField";
 import FormOptions from "./FormOptions";
-import ParticipationData from "./ParticipationData";
+import { TEXT_FIELD } from "./FormOptions/AmendProtocol";
 import ProposalTypeSelect from "./ProposalTypeSelect";
-import TitleField from "./TitleField";
-import WhenField from "./WhenField";
-
-const buildProposalOptions = (type: ProposalType, values: FormValues): ProposalOptions => {
-  const [year, month, day] = values["Date"].split("-").map(el => parseInt(el, 10));
-  const [hour, minute] = values["Time"].split(":").map(el => parseInt(el, 10));
-
-  const title = values["Title"];
-  const description = values["Description"];
-  const startTime = new Date(year, month - 1, day, hour, minute);
-  //TODO Get Election Rule from values when the UI for selecting committee is done
-  const electionRuleId = 1;
-
-  const commonOptions = {
-    title,
-    description,
-    startTime,
-    electionRuleId,
-  };
+import TitleField, { TITLE_FIELD } from "./TitleField";
+import WhenField, { DATE_FIELD, TIME_FIELD } from "./WhenField";
+
+const getElectionRules = async (governor: Governor): Promise<readonly ElectionRule[]> => {
+  if (!governor) throw new Error("Governor not set in store. This is a bug.");
```

Comment:
We should not create runtime tests which double check what we know from the type system. `governor: Governor` ensures that is set.